src/masternode.cpp

```diff
@@ -189,14 +193,26 @@ void CMasternode::Check(bool forceCheck)
             TRY_LOCK(cs_main, lockMain);
             if(!lockMain) return;
 
-            if(!AcceptToMemoryPool(mempool, state, CTransaction(tx), false, NULL, false, true, true)){
-                activeState = MASTERNODE_VIN_SPENT;
+            if(!AcceptToMemoryPool(mempool, state, CTransaction(tx), false, NULL, false, true, true)) {
+                nActiveState = MASTERNODE_OUTPOINT_SPENT;
                 return;
             }
         }
     }
 
-    activeState = MASTERNODE_ENABLED; // OK
+    nActiveState = MASTERNODE_ENABLED; // OK
+}
+
+std::string CMasternode::GetStatus()
+{
+    switch(nActiveState) {
+        case CMasternode::MASTERNODE_PRE_ENABLED:       return "PRE_ENABLED";
+        case CMasternode::MASTERNODE_ENABLED:           return "ENABLED";
+        case CMasternode::MASTERNODE_EXPIRED:           return "EXPIRED";
+        case CMasternode::MASTERNODE_OUTPOINT_SPENT:    return "OUTPOINT_SPENT";
```

Comment:
Yep, kind of... we however (slowly) starting to name things more or less right way even in user facing pieces. Few steps where already taken in 12.0 e.g. docs "collateral output transaction id", "collateral output index" https://github.com/dashpay/dash/blob/master/doc/guide-startmany.md https://github.com/dashpay/dash/blob/master/doc/masternode_conf.md (technically "output" is not the right name either but it was a huge step from "input" :) ). I think moving down this road is a good thing.